### ISSUE TITLE
Warning after update to HA 2023.2.0b0

### ISSUE THREAD

**Issue by dreimer1986** [home-assistant/android#3265]
**Home Assistant Android version:**
beta-3112-0a257070

**Android version:**
13

**Phone model:**
Pixel 7 and Surface Duo

**Home Assistant version:**
2023.2.0b0

**Last working Home Assistant release (if known):**
2023.1.x

**Description of problem:**
A new warning shows up telling me to report it to the author of the integration.

```
Logger: homeassistant.components.sensor
Source: components/sensor/__init__.py:601
Integration: Sensor ([documentation](https://rc.home-assistant.io/integrations/sensor), [issues](https://github.com/home-assistant/home-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+sensor%22))
First occurred: 22:32:31 (2 occurrences)
Last logged: 22:32:31

Sensor sensor.surface_duo_high_accuracy_update_interval has device class None, state class None and unit seconds thus indicating it has a numeric value; however, it has the non-numeric value: (<class 'str'>); Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mobile_app%22
Sensor sensor.pixel_7_pro_high_accuracy_update_interval has device class None, state class None and unit seconds thus indicating it has a numeric value; however, it has the non-numeric value: (<class 'str'>); Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mobile_app%22
```

**Traceback (if applicable, to get the logs you may refer to: https://companion.home-assistant.io/docs/troubleshooting/faqs/#android-crash-logs):**

```

```

**Screenshot of problem:**

**Additional information:**

**Comment by dshokouhi** [comment]
Loaded up latest app version and latest HA core beta release, enabled the high accuracy update interval sensor as well as other location sensors and high accuracy mode. I am unable to reproduce this error for any of our sensors.

**Comment by dshokouhi** [comment]
Ok I see now looks like this issue occurs when we send a blank state with a unit of measurement.

```
2023-01-25 14:19:08.377 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Pixel 7 Pro for type update_sensor_states: [{'unique_id': 'sleep_confidence', 'state': 15, 'type': 'sensor', 'icon': 'mdi:sleep', 'attributes': {'light': 4, 'motion': 6, 'timestamp': 1674685147000}}, {'unique_id': 'sleep_segment', 'state': '', 'type': 'sensor', 'icon': 'mdi:sleep', 'attributes': {}}, {'unique_id': 'app_rx_gb', 'state': 0.002, 'type': 'sensor', 'icon': 'mdi:radio-tower', 'attributes': {}}, {'unique_id': 'app_tx_gb', 'state': 0.0003, 'type': 'sensor', 'icon': 'mdi:radio-tower', 'attributes': {}}, {'unique_id': 'app_memory', 'state': 0.058, 'type': 'sensor', 'icon': 'mdi:memory', 'attributes': {'free_memory': 0.012, 'total_memory': 0.07}}, {'unique_id': 'app_importance', 'state': 'foreground_service', 'type': 'sensor', 'icon': 'mdi:android', 'attributes': {}}, {'unique_id': 'battery_temperature', 'state': 26.6, 'type': 'sensor', 'icon': 'mdi:battery', 'attributes': {}}, {'unique_id': 'battery_power', 'state': -2.69, 'type': 'sensor', 'icon': 'mdi:battery-minus', 'attributes': {'current': -0.647187, 'voltage': 4.158}}]
2023-01-25 14:19:08.380 WARNING (MainThread) [homeassistant.components.sensor] Sensor sensor.pixel_7_pro_sleep_segment has device class None, state class None and unit ms thus indicating it has a numeric value; however, it has the non-numeric value:  (<class 'str'>); Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mobile_app%22
```

we should probably avoid sending a blank state right? :)

**Comment by dshokouhi** [comment]
yup it keeps trying to update that same sensor as well with a blank state

**Comment by dshokouhi** [comment]
also occurs during sensor registration.

```
2023-01-25 14:37:16.978 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Pixel 7 Pro for type register_sensor: {'unique_id': 'high_accuracy_update_interval', 'state': '', 'type': 'sensor', 'icon': 'mdi:timer', 'attributes': {}, 'name': 'High Accuracy Update Interval', 'unit_of_measurement': 'seconds', 'entity_category': 'diagnostic', 'disabled': False}
2023-01-25 14:37:16.979 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Re-register for Pixel 7 Pro of existing sensor high_accuracy_update_interval
2023-01-25 14:37:46.991 WARNING (MainThread) [homeassistant.components.sensor] Sensor sensor.pixel_7_pro_high_accuracy_update_interval has device class None, state class None and unit seconds thus indicating it has a numeric value; however, it has the non-numeric value:  (<class 'str'>); Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mobile_app%22
```

**Comment by dshokouhi** [comment]
```
2023-01-25 14:45:21.853 DEBUG (MainThread) [homeassistant.components.mobile_app.webhook] Received webhook payload from Pixel 7 Pro for type register_sensor: {'unique_id': 'high_accuracy_update_interval', 'state': 5, 'type': 'sensor', 'icon': 'mdi:timer', 'attributes': {}, 'name': 'High Accuracy Update Interval', 'unit_of_measurement': 'seconds', 'entity_category': 'diagnostic', 'disabled': False}
2023-01-25 14:45:21.856 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.mobile_app entity: sensor.pixel_7_pro_high_accuracy_update_interval
2023-01-25 14:45:21.857 WARNING (MainThread) [homeassistant.components.sensor] Sensor sensor.pixel_7_pro_high_accuracy_update_interval has device class None, state class None and unit seconds thus indicating it has a numeric value; however, it has the non-numeric value:  (<class 'str'>); Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/home-assistant/core/issues?q=is%3Aopen+is%3Aissue+label%3A%22integration%3A+mobile_app%22
```

must not like the units then?

**Comment by dshokouhi** [comment]
Looks like this issue maybe core related. For now lets close in favor of https://github.com/home-assistant/core/issues/85824

**Comment by jpelgrom** [comment]
Re-opening for now because the app can fix sending an empty string as a state which gives the error `Sensor sensor.pixeldev_high_accuracy_update_interval has device class None, state class None and unit seconds thus indicating it has a numeric value; however, it has the non-numeric value:  (<class 'str'>)`. PR submitted.

A numeric value [being converted to a string](https://github.com/home-assistant/android/issues/3265#issuecomment-1404324843) was fixed in home-assistant/core#86965
